debian/patches/0002-Add-buildall.sh.patch

```diff
@@ -0,0 +1,155 @@
+From: john spurling <john.spurling@reddit.com>
+Date: Tue, 10 May 2016 19:33:09 +0000
+Subject: Add buildall.sh
+
+---
+ buildall.sh | 140 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+ 1 file changed, 140 insertions(+)
+ create mode 100755 buildall.sh
+
+diff --git a/buildall.sh b/buildall.sh
+new file mode 100755
+index 0000000..528e265
+--- /dev/null
++++ b/buildall.sh
+@@ -0,0 +1,140 @@
++#!/bin/bash
++
++REPO_BASE_DIR=`pwd`
++MCROUTER_DIR=$REPO_BASE_DIR/mcrouter
++BUILDROOT=$REPO_BASE_DIR/build
++PKG_DIR=$BUILDROOT/pkgs
++INSTALL_DIR=$BUILDROOT/install
++INSTALL_AUX_DIR=$INSTALL_DIR/aux
++
++if [ ! -d $PKG_DIR ]; then
++    mkdir -p $PKG_DIR
++fi
++
++if [[ ! -d $INSTALL_AUX_DIR ]]; then
++    mkdir -p $INSTALL_AUX_DIR
++fi
++
++export LDFLAGS="-ljemalloc $LDFLAGS"
++
++### build folly
++
++# requires: libdouble-conversion-dev
++
++cd "$PKG_DIR" || die "cd $PKG_DIR fail"
++
++if [[ ! -d folly ]]; then
++    die "folly source dir missing"
++fi
++
++# clone folly into $PKG_DIR
++# if [[ ! -d folly ]]; then
++#   git clone https://github.com/facebook/folly
```

Comment:
It doesn't need network access: I get around this by putting dependencies in git submodules.